pbh-verifier/src/helpers/PBHExternalNullifier.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import "@BokkyPooBahsDateTimeLibrary/BokkyPooBahsDateTimeLibrary.sol";
+
+/// @title PBHExternalNullifierLib
+/// @notice Library for encoding, decoding, and verifying PBH external nullifiers.
+///         External nullifiers are used to uniquely identify actions or events 
+///         within a specific year and month using a nonce.
+/// @dev Utilizes `PBHExternalNullifier` as a custom type for encoded nullifiers.
+/// @dev The encoding format is as follows:
+///      - Bits 32-255: Empty
+///      - Bits 16-31: Year
+///      - Bits 8-15: Month
+///      - Bits 0-7: Nonce
+library PBHExternalNullifier {
+    /// @notice Thrown when the provided external nullifier year doesn't
+    /// match the current year
+    error InvalidExternalNullifierYear();
+    
+    /// @notice Thrown when the provided external nullifier month doesn't
+    /// match the current month
+    error InvalidExternalNullifierMonth();
+    
+    /// @notice Thrown when the provided external 
+    /// nullifier pbhNonce >= numPbhPerMonth
+    error InvalidPbhNonce();
+
+    /// @notice Encodes a PBH external nullifier using the provided year, month, and nonce.
+    /// @param pbhNonce An 8-bit nonce value (0-255) used to uniquely identify the nullifier within a month.
+    /// @param month An 8-bit 1-indexed value representing the month (1-12).
+    /// @param year A 16-bit value representing the year (e.g., 2024).
+    /// @return The encoded PBHExternalNullifier.
+    function encode(uint8 pbhNonce, uint8 month, uint16 year) internal pure returns (uint256) {
+        require(month > 0 && month < 13, InvalidExternalNullifierMonth());
+        require(year <= 9999, InvalidExternalNullifierYear());
```

Comment:
Nit: You can use `type(uint16).max` here.

`require(year <= type(uint16).max, InvalidExternalNullifierYear());`

Reply:
isn't this a tautology ahah? a uint16 has to be <= type(uint16).max :)

Reply:
Why does it even need to be `<=` 9999? 😅